test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/ViewFeatures/ViewDataDictionaryTest.cs

```diff
@@ -253,6 +315,23 @@ public void CopyConstructorDoesNotThrowOnNullModel()
             Assert.IsType<CopyOnWriteDictionary<string, object>>(viewData.Data);
         }
 
+        [Fact]
+        public void CopyConstructor_FromValueType_DoesNotThrowOnNullModel()
+        {
+            // Arrange
+            var metadataProvider = new EmptyModelMetadataProvider();
+            var source = new TestViewDataDictionary(metadataProvider, typeof(int));
+
+            // Act
+            var viewData = new ViewDataDictionary(source, model: null);
+
+            // Assert
+            Assert.Null(viewData.Model);
+            Assert.NotNull(viewData.ModelMetadata);
+            Assert.Same(source.ModelMetadata, viewData.ModelMetadata);
```

Comment:
This assertion fails w/o the `src` changes. `viewData.ModelMetadata.ModelType` is `typeof(object)`.